### ISSUE TITLE
I also have a problem when i refresh the page

### ISSUE THREAD

**Issue by Haven09** [Cogitoergo/ccms-front#1]
I saw a issue what you have asked about Nuxt.js.
[https://github.com/nuxt/nuxt.js/issues/2680](https://github.com/nuxt/nuxt.js/issues/2680)

Does this project fix problem about using axios with token when you refresh your webpage?

**Comment by Cogitoergo** [comment]
```
// axios.js (Plugin)
export default ({ store }) => {
  axios.defaults.baseURL = {API endpoint}

  if (process.server) {
    return
  }

  axios.interceptors.request.use(request => {
    request.baseURL = {API endpoint}
    
    // Get token from auth.js store 
    const token = store.getters['auth/token']
    
    // Update token axios header
    if (token) {
      request.headers.common['Authorization'] = `Bearer ${token}`
    }
    return request
  })
})
```

This is all You need. ;)

**Comment by Haven09** [comment]
Thanks! I solved this issue,but i have another problem.
My web page get data from a java server,i want to save my common data of menus in express-session after i login successfully.
I wrote a asyncData in the next page like this
![image](https://user-images.githubusercontent.com/19543302/37693505-0a132aba-2cfb-11e8-8b54-00c4af0cb315.png)
but it didn't work, the req always was null.
Whether the problem happened because of my web page got data from a java server?It made me can't get req

**Comment by Haven09** [comment]
Thanks! I solved this issue,but i have another problem.
My web page get data from a java server,i want to save my common data of menus in express-session after i login successfully.
I wrote a asyncData method in the next page like this
![image](https://user-images.githubusercontent.com/19543302/37697053-8e43ca96-2d15-11e8-815e-bc2bb975d560.png)

but it didn't work, the req always was null.
Whether the problem happened because of my web page got data from a java server?It made me can't get req
